theano/gpuarray/opt.py

```diff
@@ -318,7 +318,20 @@ def apply(self, fgraph):
                     context_name = i.type.context_name
                     move_to_GPU = True
                     break
-
+            if (not move_to_GPU and
+                    isinstance(node.op, (theano.tensor.Alloc,
```

Comment:
I think this approximation can be extended to SubTensor as well , because we don't apply optimization if we have one client. What do you think ?